src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -225,6 +225,12 @@ public void ModifyEntriesFromPreviousTable(NodeStateTable<T> previousTable, Immu
                 }
 
                 _states.Add(modifiedEntries.ToImmutableAndFree());
+                return true;
+            }
+
+            public void AddEntries(ImmutableArray<T> values, EntryState state)
+            {
+                _states.Add(values.SelectAsArray(v => (v, state)));
```

Comment:
>values.SelectAsArray(v => (v, state)) [](http://example.com/codeflow?start=28&length=37)

Consider using `SelectAsArray()` overload that takes an extra `arg` parameter to avoid the closure class: `values.SelectAsArray((v, state) => (v, state), state)`